robot.py

```diff
@@ -44,42 +46,37 @@ def createObjects(self):
         self.spinner_solenoid = wpilib.DoubleSolenoid(2, 3)
         self.colour_sensor = rev.color.ColorSensorV3(wpilib.I2C.Port.kOnboard)
 
-    def teleopInit(self):
+    def teleopInit(self) -> None:
         """Executed at the start of teleop mode"""
 
-    def teleopPeriodic(self):
+    def teleopPeriodic(self) -> None:
         """Executed every cycle"""
-        outer_throttle = ((-self.joystick_left.getThrottle() + 1) / 2) * 5000
-        inner_throttle = -((-self.joystick_right.getThrottle() + 1) / 2) * 5000
-
-        self.shooter.set_motor_rpm(outer_throttle, inner_throttle)
-
-        wpilib.SmartDashboard.putNumber("outerError", self.shooter.get_outer_error())
-        wpilib.SmartDashboard.putNumber("centreError", self.shooter.get_centre_error())
-
-        wpilib.SmartDashboard.putNumber("outerVelocity", outer_throttle)
-        wpilib.SmartDashboard.putNumber("centreVelocity", inner_throttle)
-
-        if self.joystick_left.getRawButtonPressed(11):
-            self.loading_piston.set(wpilib.DoubleSolenoid.Value.kForward)
-        if self.joystick_left.getRawButtonPressed(12):
-            self.loading_piston.set(wpilib.DoubleSolenoid.Value.kReverse)
-
+        self.handle_shooter_inputs(self.joystick_left)
         self.handle_indexer_inputs(self.joystick_left)
         self.handle_spinner_inputs(self.spinner_joystick)
-
-    def handle_indexer_inputs(self, joystick):
+        self.send_shooter_values()
+
+    def handle_shooter_inputs(self, joystick: wpilib.Joystick) -> None:
```

Comment:
This level of abstraction seems vague. If you want to do it like this then maybe two methods for handling the piston and then handling the flywheels?